File: packages/plinth-swift/Sources/PlinthSwift/SessionMeta.swift
import Foundation

public struct SessionMeta: Codable {
    public var video: VideoMetadata
    public var client: ClientMetadata
    public var sdk: SdkMetadata

    public init(video: VideoMetadata, client: ClientMetadata, sdk: SdkMetadata) {
        self.video = video
        self.client = client
        self.sdk = sdk
    }
}

public struct VideoMetadata: Codable {
    public var id: String
    public var title: String?

    public init(id: String, title: String? = nil) {
        self.id = id
        self.title = title
    }
}

public struct ClientMetadata: Codable {
    public var userAgent: String

    public init(userAgent: String) {
        self.userAgent = userAgent
    }

    enum CodingKeys: String, CodingKey {
        case userAgent = "user_agent"
    }
}

public struct SdkMetadata: Codable {
    public var apiVersion: Int
    public var core: SdkComponent
    public var framework: SdkComponent
    public var player: SdkComponent

    public init(apiVersion: Int, core: SdkComponent, framework: SdkComponent, player: SdkComponent) {
        self.apiVersion = apiVersion
        self.core = core
        self.framework = framework
        self.player = player
    }

    enum CodingKeys: String, CodingKey {
        case apiVersion = "api_version"
        case core, framework, player
    }
}

public struct SdkComponent: Codable {
    public var name: String
    public var version: String

    public init(name: String, version: String) {
        self.name = name
        self.version = version
    }
}

File: packages/android/plinth-android/src/main/java/io/plinth/android/SessionMeta.kt
package io.plinth.android

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class SessionMeta(
    val video: VideoMetadata,
    val client: ClientMetadata,
    val sdk: SdkMetadata,
)

@Serializable
data class VideoMetadata(
    val id: String,
    val title: String? = null,
)

@Serializable
data class ClientMetadata(
    @SerialName("user_agent") val userAgent: String,
)

@Serializable
data class SdkMetadata(
    @SerialName("api_version") val apiVersion: Int,
    val core: SdkComponent,
    val framework: SdkComponent,
    val player: SdkComponent,
)

@Serializable
data class SdkComponent(
    val name: String,
    val version: String,
)
